src/p4est_io.c

```diff
@@ -2043,12 +2077,38 @@ p4est_file_read_p4est (p4est_file_context_t * fc, p4est_connectivity_t * conn,
     goto p4est_read_file_p4est_end;
   }
 
+  pertree = (p4est_gloidx_t *) pertree_arr.array;
+  /* check the read pertree array */
+  if (pertree[0] != 0) {
+    *errcode = P4EST_FILE_ERR_P4EST;
+    sc_array_reset (&pertree_arr);
+    P4EST_FILE_CHECK_NULL (*errcode, fc,
+                           P4EST_STRING "_file_read_" P4EST_STRING, errcode);
+  }
+  for (jt = 0; jt < conn->num_trees; ++jt) {
+    if (!(pertree[jt] <= pertree[jt + 1])) {
+      *errcode = P4EST_FILE_ERR_P4EST;
+      sc_array_reset (&pertree_arr);
+      P4EST_FILE_CHECK_NULL (*errcode, fc,
```

Comment:
We jump out of a loop by goto, this seems fine (?).

Reply:
This macro `P4EST_FILE_CHECK_NULL` is not a an actual goto but a clean up and error code processing followed by a `return NULL`.
Therefore, I think this clean up, error code processing and return in the loop should be fine.

The error macros only set goto if they are dedicated to be called on only one rank since then we need to synchronize at the first collective code place.